sdgx/data_processors/base.py

```diff
@@ -1,16 +1,25 @@
 from __future__ import annotations
 
+from typing import Any, Dict
+
 import pandas as pd
 
 from sdgx.data_models.metadata import Metadata
+from sdgx.exceptions import SynthesizerProcessorError
 
 
 class DataProcessor:
     """
     Base class for data processors.
     """
 
-    def fit(self, metadata: Metadata | None = None):
+    fitted = False
+
+    def check_fitted(self):
+        if not self.fitted:
+            raise SynthesizerProcessorError("Processor NOT fitted.")
+
+    def fit(self, metadata: Metadata | None = None, **kwargs: Dict[str, Any]):
```

Comment:
```suggestion
    def fit(self, metadata: Metadata | None = None, **kwargs: Dict[str, Any]):
        self._fit(metadata, **kwargs)
        self.fitted = True
    def _fit(self, metadata: Metadata | None = None, **kwargs: Dict[str, Any]):
        raise NotImplementedError
```

This way the subclass won't forget to set the fitted.